From cfe30aa1b02ee3359356c62ab98b9ec26309cdd2 Mon Sep 17 00:00:00 2001
From: Jarod Wilson <jarod@redhat.com>
Date: Thu, 22 Jan 2015 19:35:44 -0500
Subject: [redhat] kernel.spec: neuter kernel-devel rpm virtual Provides

Message-id: <20150122193544.GF10359@redhat.com>
Patchwork-id: 103349
O-Subject: [RHEL7 PATCH] redhat/kernel.spec: neuter kernel-devel rpm virtual Provides
Bugzilla: 1164344
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Aristeu Rozanski <aris@redhat.com>

Bugzilla: 1164344 - package kernel-debug-devel alleges to provide
kernel-devel, which DKMS needs but kernel-debug-devel does not actually
satisfy

Yum recently spawned full support for versioned virtual provides in
depsolving. This is a good thing, but has led to some ancient cruft biting
us in the butt. The %kernel_devel_package() macro has a virtual Provides:
of kernel-devel, which means kernel-debug-devel also Provides:
kernel-devel, and now that virtual Provides: are equivalent to Provides,
kernel-devel and kernel-debug-devel provide the exact same thing, and yum
decides which one "wins" based on the nvr, and kernel-devel =
N-V-R.debug wins over kernel-devel = N-V-R, because "R.debug" is a higher
rpm release number than "R" alone.

I believe we can fix this for the most common case by simply removing this
virtual Provides:, but this will make things possibly slightly less
convenient for someone that actually wanted kernel-debug-devel. However, I
believe the remaining virtual provides have enough options provided for
someone who wants to make a package that builds for kernel-debug-devel...

Anyway, this is causing some of the kernel module packages we build
internally in brew to fail to build, as they're trying to build for kernel
and the proper kernel-devel isn't getting installed. This should fix that.

---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/redhat/kernel.spec.template b/redhat/kernel.spec.template
index 0350b56..a0333af 100644
--- a/redhat/kernel.spec.template
+++ b/redhat/kernel.spec.template
@@ -557,7 +557,6 @@ Summary: Development package for building kernel modules to match the %{?2:%{2}
 Group: System Environment/Kernel\
 Provides: kernel%{?1:-%{1}}-devel-%{_target_cpu} = %{version}-%{release}\
 Provides: kernel-devel-%{_target_cpu} = %{version}-%{release}%{?1:.%{1}}\
-Provides: kernel-devel = %{version}-%{release}%{?1:.%{1}}\
 Provides: kernel-devel-uname-r = %{KVRA}%{?1:.%{1}}\
 AutoReqProv: no\
 Requires(pre): /usr/bin/find\
-- 
1.7.1